packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1317,10 +1332,13 @@ export class Vpc extends VpcBase {
 
     Tags.of(this).add(NAME_TAG, props.vpcName || this.node.path);
 
-    this.availabilityZones = stack.availabilityZones;
+    if (props.availabilityZones) {
+      this.availabilityZones = props.availabilityZones;
```

Comment:
I suspect there is validation that should be done on this parameter, to ensure there are no deployment time issues.

Reply:
Good idea. Done.